dbt/adapters/athena/lakeformation.py

```diff
@@ -0,0 +1,151 @@
+
+from mypy_boto3_lakeformation import LakeFormationClient
+from mypy_boto3_lakeformation.type_defs import (
+    BatchPermissionsRequestEntryTypeDef,
+    DataCellsFilterTypeDef,
+)
+from pydantic import BaseModel
+
+from dbt.adapters.athena.relation import AthenaRelation
+from dbt.events import AdapterLogger
+
+logger = AdapterLogger("AthenaLakeFormation")
+
+
+class FilterConfig(BaseModel):
+    row_filter: str
+    column_names: List[str] = []
+    principals: List[str] = []
+
+    def to_api_repr(self, catalog_id: str, database: str, table: str, name: str) -> DataCellsFilterTypeDef:
+        return {
+            "TableCatalogId": catalog_id,
+            "DatabaseName": database,
+            "TableName": table,
+            "Name": name,
+            "RowFilter": {"FilterExpression": self.row_filter},
+            "ColumnNames": self.column_names,
+            "ColumnWildcard": {"ExcludedColumnNames": []},
+        }
+
+    def to_update(self, existing: DataCellsFilterTypeDef) -> bool:
+        return self.row_filter != existing["RowFilter"]["FilterExpression"] or set(self.column_names) != set(
+            existing["ColumnNames"]
+        )
+
+
+class DataCellFiltersConfig(BaseModel):
+    enabled: bool = False
+    filters: Dict[str, FilterConfig]
+
+
+class LfGrantsConfig(BaseModel):
+    data_cell_filters: DataCellFiltersConfig
+
+
+class LfPermissions:
+    def __init__(self, catalog_id: str, relation: AthenaRelation, lf_client: LakeFormationClient) -> None:
+        self.catalog_id = catalog_id
+        self.relation = relation
+        self.database: str = relation.schema
+        self.table: str = relation.identifier
+        self.lf_client = lf_client
+
+    def get_filters(self) -> Dict[str, DataCellsFilterTypeDef]:
+        table_resource = {"CatalogId": self.catalog_id, "DatabaseName": self.database, "Name": self.table}
+        return {f["Name"]: f for f in self.lf_client.list_data_cells_filter(Table=table_resource)["DataCellsFilters"]}
+
+    def process_filters(self, config: LfGrantsConfig) -> None:
+        current_filters = self.get_filters()
+        logger.debug(f"CURRENT FILTERS: {current_filters}")
+
+        to_drop = [f for name, f in current_filters.items() if name not in config.data_cell_filters.filters]
+        if to_drop:
+            logger.debug(f"FILTERS TO DROP: {to_drop}")
+            for f in to_drop:
+                self.lf_client.delete_data_cells_filter(**f)
+        else:
+            logger.debug("NO FILTERS TO DROP FOUND")
+
+        to_add = [
+            f.to_api_repr(self.catalog_id, self.database, self.table, name)
+            for name, f in config.data_cell_filters.filters.items()
+            if name not in current_filters
+        ]
+        if to_add:
+            logger.debug(f"FILTERS TO ADD: {to_add}")
+            for f in to_add:
+                self.lf_client.create_data_cells_filter(TableData=f)
+        else:
+            logger.debug("NO FILTERS TO ADD FOUND")
```

Comment:
As far as I understand the if/else is mostly added for debugging. But how about keeping it a bit clean and remove it? 
or add simply: `logger.debug(f'to_add: {to_add}')` doing so we can remove 3 if/else statement and keep the code cleaner, but at the same time have some debugging.

Reply:
good point! BTW, python3.7 support makes me a bit upset 😄 at first iteration I used `:=` to minify code and added these `if else` statements later, will fix it

Reply:
I'm waiting for the day that dbt-core stop to support python 3.7 :D 
thanks for the change.